server/internal/infrastructure/memory/transaction.go

```diff
@@ -6,20 +6,52 @@ import (
 	"github.com/reearth/reearth-cms/server/internal/usecase/repo"
 )
 
-type Transaction struct{}
+type Transaction struct {
+	committed  int
+	beginError error
+	endError   error
+}
 
-type Tx struct{}
+type Tx struct {
+	t         *Transaction
+	committed bool
+	endError  error
+}
 
 func NewTransaction() *Transaction {
 	return &Transaction{}
 }
 
-func (*Transaction) Begin() (repo.Tx, error) {
-	return &Tx{}, nil
+func (t *Transaction) SetBeginError(err error) {
+	t.beginError = err
+}
+
+func (t *Transaction) SetEndError(err error) {
+	t.endError = err
+}
+
+func (t *Transaction) Committed() int {
+	return t.committed
 }
-func (*Tx) Commit() {
-	// do nothing
+
+func (t *Transaction) Begin() (repo.Tx, error) {
+	if t.beginError != nil {
+		return nil, t.beginError
+	}
+	return &Tx{t: t, endError: t.endError}, nil
+}
+
+func (t *Tx) Commit() {
+	t.committed = true
+	if t.t != nil {
+		t.t.committed++
+	}
 }
-func (*Tx) End(_ context.Context) error {
-	return nil
+
+func (t *Tx) End(_ context.Context) error {
+	return t.endError
```

Comment:
```suggestion
	if t.endError != nil {
		return t.endError
	}
	if t.committed && t.t != nil {
		t.t.committed++
	}
	return nil
```

https://github.com/reearth/reearth-backend/pull/155/files#diff-be2500965536f0d354f85ad68d0831cb5bd01016f7ca0d98eccaf68a04b5a160